### PR TITLE
Fix for inconsistent tests

### PR DESCRIPTION
**Pull request by rbasralian**
Improvements to the intermittently-failing tests in `test_function_genreated_table.py` (#4476)

(2/3 fixed — `test_generated_table_1trigger` still a mystery)

### CODE REVIEW

File: py/server/tests/test_function_generated_table.py [hunk 1]
```diff
@@ -39,20 +42,26 @@ def test_generated_table_1trigger(self):
...
         result_table = function_generated_table(table_generator_function, source_tables=append_only_input_table)
 
         self.assertEqual(result_table.size, 0)
 
+        print("Adding row at time: " + str(datetime.now()))
         append_only_input_table.add(new_table([string_col(name='MyStr', data=['test string'])]))
+        print("add() returned at time: " + str(datetime.now()))
         self.wait_ticking_table_update(result_table, row_count=1, timeout=30)
-        self.assertEqual(result_table.size, 1)
+        print("result_table has 1 row at time: " + str(datetime.now()))
```
Comment (jmao-denver) [inline py/server/tests/test_function_generated_table.py:56]: So the row somehow vanished. Is the table a blink table?

File: py/server/tests/test_function_generated_table.py [hunk 2]
```diff
@@ -39,20 +42,26 @@ def test_generated_table_1trigger(self):
...
 
         result_str = result_table.j_table.getColumnSource("ResultStr").get(0)
         self.assertEqual(result_str, 'test string')
```
Comment (rcaudy) [inline py/server/tests/test_function_generated_table.py:59]: This is where we were failing, right @rbasralian ? @jmao-denver or @chipkent, is it possible that this is the wrong way to compare strings? I see some stuff in the unittest.TestCase documentation about type-specific equals that was added to assertEquals in 3.1. Is that Python 3.1, or v3.1 of the testcase package, or?

File: py/server/tests/test_function_generated_table.py [hunk 3]
```diff
@@ -39,20 +42,26 @@ def test_generated_table_1trigger(self):
...
 
     def test_generated_table_2triggers(self):
+        # NOTE: This tests that both trigger tables cause the refresh function to run.
+        # It does not test updating two source tables in the same cycle (that is covered by
+        # io.deephaven.engine.table.impl.util.TestFunctionGeneratedTableFactory.testMultipleSources).
```
Comment (rcaudy) [inline py/server/tests/test_function_generated_table.py:64]: I don't see where this is tested at all. There's no "wait".

File: py/server/tests/test_function_generated_table.py (hunk 1 above)
Comment (rbasralian) [reply py/server/tests/test_function_generated_table.py:56]: nah — it's an input table, which is backed by an AppendOnlyArrayBackedMutableTable

File: py/server/tests/test_function_generated_table.py [hunk 4]
```diff
@@ -39,20 +42,26 @@ def test_generated_table_1trigger(self):
...
         self.assertEqual(result_table.size, 0)
 
+        print("Adding row at time: " + str(datetime.now()))
         append_only_input_table.add(new_table([string_col(name='MyStr', data=['test string'])]))
+        print("add() returned at time: " + str(datetime.now()))
         self.wait_ticking_table_update(result_table, row_count=1, timeout=30)
-        self.assertEqual(result_table.size, 1)
+        print("result_table has 1 row at time: " + str(datetime.now()))
 
         result_str = result_table.j_table.getColumnSource("ResultStr").get(0)
```
Comment (rcaudy) [inline py/server/tests/test_function_generated_table.py:58]: Don't do `get(0)`, use the row set.

File: py/server/tests/test_function_generated_table.py (hunk 2 above)
Comment (jmao-denver) [reply py/server/tests/test_function_generated_table.py:59]: `assertEqual` is the right way. The logged error is "None not equal to 'test string' which indicates `get(0)` returning a null value.

**Review by chipkent — APPROVED** [review]

**Review by rcaudy — APPROVED** [review]

**Pull request merged**